packages/remix-dev/vite/plugin.ts

```diff
@@ -737,6 +747,88 @@ export const remixVitePlugin: RemixVitePlugin = (options = {}) => {
           }
         };
       },
+      writeBundle: {
+        // After the SSR build is finished, we inspect the Vite manifest for
+        // the SSR build and move all server assets to client assets directory
+        async handler() {
+          if (!ssrBuildContext.isSsrBuild) {
+            return;
+          }
+
+          invariant(
+            cachedPluginConfig,
+            "Expected plugin config to be cached when writeBundle hook is called"
+          );
+
+          invariant(
+            resolvedViteConfig,
+            "Expected resolvedViteConfig to exist when writeBundle hook is called"
+          );
+
+          let { assetsBuildDirectory, serverBuildPath, rootDirectory } =
+            cachedPluginConfig;
+          let serverBuildDir = path.dirname(serverBuildPath);
+
+          let ssrViteManifest = await loadViteManifest(serverBuildDir);
+          let clientViteManifest = await loadViteManifest(assetsBuildDirectory);
+
+          let clientAssetPaths = new Set(
+            Object.values(clientViteManifest).flatMap(
+              (chunk) => chunk.assets ?? []
+            )
+          );
+
+          let ssrOnlyAssetPaths = new Set(
+            Object.values(ssrViteManifest)
+              .flatMap((chunk) => chunk.assets ?? [])
+              // Only move assets that aren't in the client build
+              .filter((asset) => !clientAssetPaths.has(asset))
+          );
+
+          let movedAssetPaths = await Promise.all(
+            Array.from(ssrOnlyAssetPaths).map(async (ssrAssetPath) => {
+              let src = path.join(serverBuildDir, ssrAssetPath);
+              let dest = path.join(assetsBuildDirectory, ssrAssetPath);
+              await fse.move(src, dest);
+              return dest;
+            })
+          );
+
+          let logger = vite.createLogger();
+
+          if (movedAssetPaths.length) {
+            logger.info(
+              [
+                "",
+                `${colors.green("✓")} ${movedAssetPaths.length} asset${
+                  movedAssetPaths.length > 1 ? "s" : ""
+                } moved from Remix server build to client assets.`,
+                ...movedAssetPaths.map((movedAssetPath) =>
+                  colors.dim(path.relative(rootDirectory, movedAssetPath))
+                ),
+              ].join("\n")
+            );
+          }
+
+          let ssrAssetsDir = path.join(
+            resolvedViteConfig.build.outDir,
+            resolvedViteConfig.build.assetsDir
+          );
+
+          if (fse.existsSync(ssrAssetsDir)) {
+            await fse.remove(ssrAssetsDir);
+            logger.info(
+              [
+                "",
+                colors.dim(
+                  `${colors.green("✓")} Cleaned Remix server assets directory.`
+                ),
```

Comment:
Maybe not intended to be `dim`?

<details><summary>with dim</summary>

![image](https://github.com/remix-run/remix/assets/4232207/76092945-bc05-4628-b39c-b83bb6178ea8)

</details>

<details><summary>without dim</summary>

![image](https://github.com/remix-run/remix/assets/4232207/1ebda188-c116-4e84-b06d-ba815ce02513)

</details>

Reply:
Spoke with @pcattori about this and we're opting to remove this line instead.

Reply:
Sounds good to me. Indeed, 1st part of the log looks good enough.